File: Examples/android/CMeSdkJava/app_rework/src/main/java/me/connect/sdk/java/sample/credentials/CredentialOffersAdapter.java
package me.connect.sdk.java.sample.credentials;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;
import java.util.List;

import me.connect.sdk.java.sample.R;
import me.connect.sdk.java.sample.db.entity.CredentialOffer;

public class CredentialOffersAdapter extends RecyclerView.Adapter<CredentialOffersAdapter.CredentialsViewHolder> {

    private List<CredentialOffer> data;
    private ItemClickListener itemClickListener;

    public CredentialOffersAdapter(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
        data = new ArrayList<>();
    }

    @NonNull
    @Override
    public CredentialsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.credentials_list_item, parent, false);
        return new CredentialsViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull CredentialsViewHolder holder, int position) {
        CredentialOffer credentialOffer = data.get(position);
        holder.text.setText(credentialOffer.name);
        holder.offers.setText(credentialOffer.attributes);
        int res;
        if (credentialOffer.accepted == null)
            res = R.drawable.question;
        else if (credentialOffer.accepted)
            res = R.drawable.yes;
        else
            res = R.drawable.no;
        holder.image.setImageResource(res);
        holder.accept.setVisibility(credentialOffer.accepted == null ? View.VISIBLE : View.GONE);
        holder.accept.setOnClickListener(view -> {
            holder.accept.setEnabled(false);
            itemClickListener.onItemClick(credentialOffer.id);
        });
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void setData(List<CredentialOffer> credentials) {
        data = credentials;
        notifyDataSetChanged();
    }

    static class CredentialsViewHolder extends RecyclerView.ViewHolder {
        public TextView text;
        public ImageView image;
        public TextView offers;
        public Button accept;

        public CredentialsViewHolder(View v) {
            super(v);
            text = v.findViewById(R.id.text);
            image = v.findViewById(R.id.image);
            offers = v.findViewById(R.id.offers);
            accept = v.findViewById(R.id.buttonaccept);
        }
    }

    public interface ItemClickListener {
        void onItemClick(int credOfferId);
    }
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/credentials/CredentialOffersAdapter.kt
package me.connect.sdk.java.samplekt.credentials

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import me.connect.sdk.java.samplekt.R
import me.connect.sdk.java.samplekt.db.entity.CredentialOffer


class CredentialOffersAdapter(private val itemClickListener: ItemClickListener) : RecyclerView.Adapter<CredentialOffersAdapter.CredentialsViewHolder>() {
    private val data = mutableListOf<CredentialOffer>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CredentialsViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.credentials_list_item, parent, false)
        return CredentialsViewHolder(v)
    }

    override fun onBindViewHolder(holder: CredentialsViewHolder, position: Int) {
        val credentialOffer= data[position]
        holder.text.text = credentialOffer.name
        holder.offers.text = credentialOffer.attributes
        val res: Int
        holder.accept.isEnabled = true
        res = when (credentialOffer.accepted) {
            null -> R.drawable.question
            true -> R.drawable.yes
            else -> R.drawable.no
        }
        holder.image.setImageResource(res)
        Glide.with(holder.logo.context).load(credentialOffer.attachConnectionLogo).into(holder.logo)
        holder.accept.visibility = if (credentialOffer.accepted == null) View.VISIBLE else View.GONE
        holder.accept.setOnClickListener {
            holder.accept.isEnabled = false
            itemClickListener.onItemClick(credentialOffer.id)
        }
    }

    override fun getItemCount(): Int = data.size

    fun setData(credentials: List<CredentialOffer>) {
        data.clear()
        data.addAll(credentials)
        notifyDataSetChanged()
    }

    class CredentialsViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var text: TextView = v.findViewById(R.id.text)
        var image: ImageView = v.findViewById(R.id.image)
        var logo: ImageView = v.findViewById(R.id.logo)
        var offers: TextView = v.findViewById(R.id.offers)
        var accept: Button = v.findViewById(R.id.buttonaccept)
    }

    interface ItemClickListener {
        fun onItemClick(credOfferId: Int)
    }
}
